File: Modulo12/Professor-Java/br/ufscar/dc/poo/Pessoa.java
package br.ufscar.dc.poo;

public class Pessoa implements Comparable<Pessoa> {

    private String nome;
    private int CPF;

    public Pessoa(String nome, int CPF) {
        this.CPF = CPF;
        this.nome = nome;
    }

    public int getCPF() {
        return CPF;
    }

    public void setCPF(int CPF) {
        this.CPF = CPF;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void imprime() {
        System.out.println("Nome: " + nome);
        System.out.println("CPF: " + CPF);
    }

    @Override
    public int compareTo(Pessoa outro) {
        if (outro == null) {
            return 1;
        } else {
            return this.nome.compareTo(outro.nome);
        }
    }
}

File: Modulo12/Professor-CSharp/Pessoa.cs
using System;
using System.Collections;

namespace poo {

    public class Pessoa : IComparable<Pessoa> {

        private string nome;
        private int CPF;

        public Pessoa(string nome, int CPF) {
            this.CPF = CPF;
            this.nome = nome;
        }

        public int getCPF() {
            return CPF;
        }

        public void setCPF(int CPF) {
            this.CPF = CPF;
        }

        public string getNome() {
            return nome;
        }

        public void setNome(string nome) {
            this.nome = nome;
        }

        public virtual void imprime() {
            Console.WriteLine("Nome: " + nome);
            Console.WriteLine("CPF: " + CPF);
        }

        public int CompareTo(Pessoa outro) { // comparação padrão => por nome
            if (outro == null)
                return 1;
            else
                return this.nome.CompareTo(outro.nome);
        }
    }

}